hypervisor/src/hypervisor/partition.rs

```diff
@@ -672,3 +674,61 @@ impl Partition {
         self.base.cgroup.rm().typ(SystemError::CGroup)
     }
 }
+
+impl PartitionConfig {
+    /// Get the path to a partition binary
+    ///
+    /// The [PartitionConfig::image] field must either:
+    /// - not contain any path separators, in which case `$PATH` is searched for
+    ///   a matching executable (like the `which` command in the shell would do)
+    /// - be an absolute path, in which case the path is used verbatim after
+    ///   verification that it:
+    ///   - exists
+    ///   - is a file
+    ///   - is executable
+    /// - be a relative path starting with `./`, in which case it is resolved
+    ///   relative to the hypervisors current workind directory
+    fn get_partition_bin(&self) -> TypedResult<PathBuf> {
+        let PartitionConfig { image, name, .. } = self;
+
+        // if image is either an absolute path or starts with ./ , it is left as is
+        let bin = if image.is_absolute() || image.starts_with(path::Component::CurDir) {
+            // verify image exists
+            if !image.exists() {
+                problem!(Panic, "partition image {image:?} does not exist");
+            } else if !image.is_file() {
+                problem!(Panic, "partition image {image:?} is not a file");
+            } else if image
+                .metadata()
+                .typ(SystemError::Panic)?
+                .permissions()
+                .mode()
+                & 0b1
```

Comment:
You might be able to get less indentations. One example would be that you could first check if the file exists and then early exit using `?`.